examples/SampleUtils.h

```diff
@@ -99,11 +99,53 @@ namespace utils {
         bool transpose = false;
         int32_t groups = 1;
         ml::InputOperandLayout inputLayout = ml::InputOperandLayout::Nchw;
-        ml::FilterOperandLayout filterLayout = ml::FilterOperandLayout::Oihw;
+        ml::Conv2dFilterOperandLayout filterLayout = ml::Conv2dFilterOperandLayout::Oihw;
         ml::Operand bias;
         ml::Operator activation;
 
         const ml::Conv2dOptions* AsPtr() {
+            if (!padding.empty()) {
+                mOptions.paddingCount = padding.size();
+                mOptions.padding = padding.data();
+            }
+            if (!strides.empty()) {
+                mOptions.stridesCount = strides.size();
+                mOptions.strides = strides.data();
+            }
+            if (!dilations.empty()) {
+                mOptions.dilationsCount = dilations.size();
+                mOptions.dilations = dilations.data();
+            }
+            mOptions.groups = groups;
+            mOptions.autoPad = autoPad;
+            mOptions.inputLayout = inputLayout;
+            mOptions.filterLayout = filterLayout;
+            mOptions.bias = bias;
+            mOptions.activation = activation;
+
+            return &mOptions;
+        }
+
+      private:
+        ml::Conv2dOptions mOptions;
+    };
+
+    struct ConvTranspose2dOptions {
```

Comment:
The struct `ConvTranspose2dOptions` and `Conv2dOptions` are similar, can we define a base template like
```
template<typename OptionType>
struct Conv2dBaseOptions {
   public:
        std::vector<int32_t> padding;
        std::vector<int32_t> strides;
        std::vector<int32_t> dilations;
        ml::AutoPad autoPad = ml::AutoPad::Explicit;
        int32_t groups = 1;
        ml::InputOperandLayout inputLayout = ml::InputOperandLayout::Nchw;
        ml::Operand bias;
        ml::Operator activation;

     const OptionType* AsPtr() {
     }

    private:
      OptionType mOptions;
}
```
Conv2d can extend it like
```
struct Conv2dOptions : pubilc ObjectBase<Conv2dBaseOptions> {
    pubilc:
       ml::Conv2dFilterOperandLayout filterLayout;
}
```
Transpose Conv2d can do like
```
struct ConvTranspose2dOptions: pubilc ObjectBase<Conv2dBaseOptions> {
    pubilc:
       ml::ConvTranspose2dFilterOperandLayout filterLayout;
        std::vector<int32_t> outputPadding;
        std::vector<int32_t> outputSizes;
}
```

Other implemenations like node/src/ops/ConvTranspose2d.cpp and src/webnn_native/ops/ConvTranspose2d.cpp also can consider this design.

Reply:
@fujunwei This is a very important reminder, I will update it in next commit.